File: GZCTF/Models/Request/Account/ClientUserInfoModel.cs
using System.ComponentModel.DataAnnotations;

namespace CTFServer.Models.Request.Account;

public class ClientUserInfoModel
{
    /// <summary>
    /// 用户ID
    /// </summary>
    public string UserId { get; set; }

    /// <summary>
    /// 用户名
    /// </summary>
    public string? UserName { get; set; }

    /// <summary>
    /// 邮箱
    /// </summary>
    public string? Email { get; set; }

    /// <summary>
    /// 签名
    /// </summary>
    public string? Bio { get; set; }

    /// <summary>
    /// 手机号码
    /// </summary>
    public string? Phone { get; set; }

    /// <summary>
    /// 头像链接
    /// </summary>
    public string? Avatar { get; set; }

    /// <summary>
    /// 当前队伍
    /// </summary>
    public int ActiveTeamId { get; set; }

    public static ClientUserInfoModel FromUserInfo(UserInfo user)
        => new()
        {
            UserId = user.Id,
            Bio = user.Bio,
            Email = user.Email,
            UserName = user.UserName,
            Phone = user.PhoneNumber,
            Avatar = $"/assets/{user.AvatarHash}/avatar",
            ActiveTeamId = user.ActiveTeamId
        };
}

File: GZCTF/ClientApp/pages/client/models/ClientUserInfoModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Role } from './Role';

export type ClientUserInfoModel = {
    /**
     * 用户ID
     */
    userId?: string | null;
    /**
     * 用户名
     */
    userName?: string | null;
    /**
     * 邮箱
     */
    email?: string | null;
    /**
     * 签名
     */
    bio?: string | null;
    /**
     * 手机号码
     */
    phone?: string | null;
    /**
     * 真实姓名
     */
    realName?: string | null;
    /**
     * 头像链接
     */
    avatar?: string | null;
    /**
     * 当前队伍
     */
    activeTeamId?: number | null;
    /**
     * 用户角色
     */
    role?: Role | null;
};
